webapp/tests/karma/ts/modules/analytics/analytics-target-aggregates-sidebar-filter-component.spec.ts

```diff
@@ -0,0 +1,144 @@
+        ],
+        declarations: [
+          AnalyticsTargetAggregatesSidebarFilterComponent,
+        ],
+        providers: [
+          provideMockStore({ selectors: mockedSelectors }),
+          { provide: ContactTypesService, useValue: contactTypesService },
+          { provide: SettingsService, useValue: settingsService },
+          { provide: TelemetryService, useValue: telemetryService },
+          { provide: UserSettingsService, useValue: userSettingsService },
+        ]
+      })
+      .compileComponents()
+      .then(() => {
+        fixture = TestBed.createComponent(AnalyticsTargetAggregatesSidebarFilterComponent);
+        component = fixture.componentInstance;
+        store = TestBed.inject(MockStore);
+        fixture.detectChanges();
+      });
+  }));
+
+  afterEach(() => {
+    store.resetSelectors();
+    sinon.restore();
+  });
+
+  it('should create component', fakeAsync(() => {
+    expect(component).to.exist;
+    expect(component.isOpen).to.be.false;
+    expect(component.error).to.be.undefined;
+  }));
+
+  it('should toggle sidebar filter', () => {
+    component.toggleSidebarFilter();
+    component.toggleSidebarFilter();
+    component.toggleSidebarFilter();
+
+    expect(globalActions.setSidebarFilter.calledThrice).to.be.true;
+    expect(globalActions.setSidebarFilter.args[0][0]).to.deep.equal({ isOpen: true });
+    expect(globalActions.setSidebarFilter.args[1][0]).to.deep.equal({ isOpen: false });
+    expect(globalActions.setSidebarFilter.args[2][0]).to.deep.equal({ isOpen: true });
+    expect(telemetryService.record.calledTwice).to.be.true;
+  });
+
+  it('should set userFacilities and selectedFacilityId when user has multiple facilities', fakeAsync(() => {
+    const userFacilities = [{ _id: 'facility_1' }, { _id: 'facility_2' }];
+    userSettingsService.getUserFacility.resolves(userFacilities);
+    userSettingsService.hasMultipleFacilities.resolves(true);
+
+    expect(component.userFacilities).to.deep.equal(userFacilities);
+    expect(component.selectedFacilityId).to.equal('facility_1');
+    expect(userSettingsService.getUserFacility.calledOnce).to.be.true;
+    expect(userSettingsService.hasMultipleFacilities.calledOnce).to.be.true;
+  }));
+
+  it('should set user facility name_key as facilityFilterLabel', fakeAsync(() => {
+    contactTypesService.getTypeId.returns('district_hospital');
+    component.userFacilities = [{ _id: 'facility_1', type: 'district_hospital' }];
+
+    expect(component.facilityFilterLabel).to.equal('District Hospital');
+    expect(settingsService.get.calledOnce).to.be.true;
+    expect(contactTypesService.getTypeId.calledOnce).to.be.true;
+  }));
+
+  it('should set error and default facilityFilterLabel when settings service fails', fakeAsync(() => {
+    const DEFAULT_FACILITY_LABEL = 'Facility';
+    settingsService.get.rejects({ some: 'err' });
+
+    component.ngOnInit();
+    flush();
+
+    expect(component.facilityFilterLabel).to.equal(DEFAULT_FACILITY_LABEL);
```

Comment:
Same here, assert functions are called with the correct amount of time (call count)